tests/test_flows.py

```diff
@@ -707,30 +707,28 @@ def test_link_metric_query(self):
         query = self.check_cmdu_type_single("link metric query", 0x0005,
                                             env.controller.mac, env.agents[0].mac,
                                             mid)
+        assert query
 
-        if query:
-            query_tlv = self.check_cmdu_has_tlv_single(query, 0x08)
+        query_tlv = self.check_cmdu_has_tlv_single(query, 0x08)
+        assert query_tlv
 
-        if query and query_tlv:
-            debug("Checking query type and queried metrics are correct")
-            try:
-                if int(query_tlv.link_metric_query_type) != 0x00:
-                    self.fail("Query type is not 'All neighbors'")
-            except (AttributeError, ValueError):
-                self.fail("Query type is not recognized")
-            try:
-                if int(query_tlv.link_metrics_requested) != 0x02:
-                    self.fail("Metrics for both Tx and Rx is not requested")
-            except (AttributeError, ValueError):
-                self.fail("Requested metrics are not recognized")
+        debug("Checking query type and queried metrics are correct")
+        assert query_tlv.link_metric_query_type, "Query type is not recognized"
```

Comment:
This is not needed. We can pretty much assume that it's there, otherwise the tlv wouldn't exist to begin with. And if it doesn't, the following statement will simply raise an AttributeError.

Reply:
I do that to prevent the AttributeError, test should only fail with AssertionError I think, otherwise we can't discern if the test failed or the test code failed.

Reply:
It doesn't actually prevent the attribute error - the assert will never even trigger if the attribute doesn't exist. You'd need
```
assert hasattr(query_tlv, "link_metric_query_type")
```

However, I think it is much more convenient if we can just access attributes without explicit asserts. In this case, we could actually implement a `__getattr__` function on TlvStruct that asserts.